tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_store_utils.ts

```diff
@@ -34,3 +34,54 @@ export function findFileIndex(
       item.file_path === fileSpec.file_path
   );
 }
+
+/**
+ * Find the first range (with begin and end properties) in an array of ranges
+ * that equals `[begin, end)`.
+ *
+ * @param ranges The ranges to search in.
+ * @param begin The begin of the range to search for (inclusive).
+ * @param end The end of the range to search for (exclusive).
+ * @returns Index (>=0) if found. -1 if not found.
+ */
+export function findBeginEndRangeIndex(
+  ranges: Array<{begin: number; end: number}>,
+  begin: number,
+  end: number
+): number {
+  if (begin >= end) {
+    throw new Error(
+      `Expected begin to be less than end, ` +
+        `but got begin=${begin}, end=${end}`
+    );
+  }
+  return ranges.findIndex(
+    (range) => range.begin === begin && range.end === end
+  );
+}
+
+/**
+ * Detemrine if an array of ranges contains a range that strictly includes
+ * `[begin, end)`.
+ *
+ * @param ranges The ranges to search in.
+ * @param begin The begin of the range to search for (inclusive).
+ * @param end The end of the range to search for (exclusive).
+ * @returns `True` if and only if one of the ranges of `ranges` strictly
+ *   includes `[begin, end)`.
```

Comment:
AFAICT, it's both inclusive, right? i.e., `[begin, end]`

Reply:
Confirming the ranges are all left-inclusive-right-exclusive. This has been consistent in debugger-v2 code, and consistent with Python indexing as well. Keep in mind that the code `range.begin >= begin && range.end <= end` below is *not* about comparing a range with indices, but about comparing two ranges.

To give an example, if you have `range = [0, 3)` here and `begin = 0, end = 3`, this function should (and does) return
`true`.